edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -15,10 +15,24 @@ public class DockerConfig : IEquatable<DockerConfig>
     {
         // This is not the actual docker image regex, but a less strict version.
         const string ImageRegexPattern = @"^(?<repo>([^/]*/)*)(?<image>[^/:]+)(?<tag>:[^/:]+)?$";
```

Comment:
sidetrack: why we need to split into 3 parts to validate the image address.  It is better to split into 2 parts - first part is full URL and 2nd part is image tag.  Then first part can be validated from Uri.IsWellFormedUriString.

Reply:
I agree. Repo is not used, however I am in favor of not touching since any change introduces a risk.